media-sound/mixxx/mixxx-9999.ebuild

```
@@ -14,114 +14,141 @@ if [[ ${PV} == *9999 ]] ; then
 	fi
 	EGIT_REPO_URI="https://github.com/mixxxdj/${PN}.git"
 else
-	SRC_URI="https://github.com/mixxxdj/${PN}/archive/release-${PV}.tar.gz -> ${P}.tar.gz"
-	S="${WORKDIR}/${PN}-release-${PV}"
+	SRC_URI="https://github.com/mixxxdj/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
 	KEYWORDS="~amd64 ~x86"
 fi
 
 LICENSE="GPL-2"
 SLOT="0"
-IUSE="aac doc ffmpeg hid keyfinder lv2 mp3 mp4 opus qtkeychain shout wavpack"
+IUSE="aac ffmpeg keyfinder lv2 midi modplug mp3 mp4 opus qml qtkeychain"
+IUSE+=" rubberband shout test upower wavpack"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	qtkeychain? ( shout )
+	test? ( aac ffmpeg midi mp3 opus rubberband )
+"
 
-RDEPEND="
+COMMON_DEPEND="
 	dev-db/sqlite
-	dev-libs/glib:2
+	dev-libs/hidapi
 	dev-libs/protobuf:=
-	dev-qt/qtconcurrent:5
-	dev-qt/qtcore:5
-	dev-qt/qtdbus:5
-	dev-qt/qtgui:5
-	dev-qt/qtnetwork:5
-	dev-qt/qtopengl:5
-	dev-qt/qtscript:5[scripttools]
-	dev-qt/qtsql:5
-	dev-qt/qtsvg:5
-	dev-qt/qtwidgets:5
-	dev-qt/qtx11extras:5
-	dev-qt/qtxml:5
+	dev-qt/qt5compat:6
+	dev-qt/qtbase:6[concurrent,dbus,gui,icu,network,opengl,sql,sqlite,widgets,xml,X]
+	dev-qt/qtdeclarative:6
+	dev-qt/qtshadertools:6
+	dev-qt/qtsvg:6
 	media-libs/chromaprint
 	media-libs/flac:=
 	media-libs/libebur128
-	media-libs/libid3tag:=
+	media-libs/libglvnd[X]
 	media-libs/libogg
 	media-libs/libsndfile
 	media-libs/libsoundtouch
 	media-libs/libvorbis
 	media-libs/portaudio
-	media-libs/portmidi
-	media-libs/rubberband
 	media-libs/taglib:=
-	media-libs/vamp-plugin-sdk
-	sci-libs/fftw:3.0=
-	sys-power/upower
+	media-sound/lame
 	virtual/glu
 	virtual/libusb:1
-	virtual/opengl
 	virtual/udev
 	x11-libs/libX11
-	aac? (
-		media-libs/faad2
-		media-libs/libmp4v2
-	)
+	aac? ( media-libs/libmp4v2 )
 	ffmpeg? ( media-video/ffmpeg:= )
-	hid? ( dev-libs/hidapi )
 	keyfinder? ( media-libs/libkeyfinder )
 	lv2? ( media-libs/lilv )
-	mp3? ( media-libs/libmad )
-	mp4? ( media-libs/libmp4v2:= )
-	opus? (	media-libs/opusfile )
-	qtkeychain? ( dev-libs/qtkeychain:=[qt5(+)] )
-	shout? ( >=media-libs/libshout-2.4.5 )
+	midi? ( media-libs/portmidi )
+	modplug? ( media-libs/libmodplug )
+	mp3? (
+		media-libs/libid3tag:=
+		media-libs/libmad
+	)
+	mp4? ( media-libs/libmp4v2 )
+	opus? (
+		media-libs/opus
+		media-libs/opusfile
+	)
+	qtkeychain? ( dev-libs/qtkeychain:=[qt6(-)] )
+	rubberband? ( media-libs/rubberband )
+	shout? ( dev-libs/openssl:= )
+	upower? (
+		dev-libs/glib
+		sys-power/upower
+	)
 	wavpack? ( media-sound/wavpack )
 "
-DEPEND="${RDEPEND}
+RDEPEND="${COMMON_DEPEND}
+	aac? ( media-libs/faad2 )
+"
+DEPEND="${COMMON_DEPEND}
 	dev-cpp/ms-gsl
+	test? ( media-libs/faad2 )
```

```suggestion
	test? ( media-libs/faad2 )
```

For simplicity I think its to have fine to have media-libs/faad2 is in COMMON_DEPEND. It isn't going to cause harm.

With QA-VDB comment is was just explaining the output of qa-vdb really.

---

You mean as initial commit ? ...

---

I'm refering to this https://github.com/gentoo/gentoo/pull/40226#discussion_r1922593544. Like it would be more correct to have it out of DEPEND, but it doesn't cause harm if is there. Especially if tests need it in DEPEND again.